nautobot/extras/jobs.py

```diff
@@ -123,26 +131,21 @@ def __call__(self, *args, **kwargs):
         self.active_test = "run"
         context_class = JobHookChangeContext if isinstance(self, JobHookReceiver) else JobChangeContext
         change_context = context_class(user=self.user, context_detail=self.class_path)
-        # TODO: re-add profiling feature
-        # if profile:
-        #     import cProfile
-
-        #     # TODO: This should probably be available as a file download rather than dumped to the hard drive.
-        #     # Pending this: https://github.com/nautobot/nautobot/issues/3352
-        #     profiling_path = f"/tmp/nautobot-jobresult-{self.request.id}.pstats"
-
-        #     # TODO: Context manager for this is added in 3.8
-        #     profiler = cProfile.Profile()
-        #     profiler.enable()
-        #     output = job.run(data=data, commit=commit)
-        #     profiler.disable()
-        #     profiler.dump_stats(profiling_path)
-        #     job.log_info(obj=None, message=f"Wrote profiling information to {profiling_path}.")
-        # else:
-        #     output = job.run(data=data, commit=commit)
 
         with change_logging(change_context):
```

Comment:
TODO: We're deserializing the `User` and `JobResult` in the `BaseJob.__call__` method. We'll need to see if this is pre or post-fork to find out if we'll run into the same issue that was fixed in #3105

Reply:
Looks ok. Also appears to run synchronously in the [celery source](https://github.com/celery/celery/blob/v5.2.7/celery/app/trace.py#L434-L545)

| method | pid | thread id |
| -------- | ---- | ------- |
| before_start | 5937 | 140496557238080 |
| __call__ | 5937 | 140496557238080 |
| after_return | 5937 | 140496557238080 |